experiments/operational_domain/operational_domain_siqad.cpp

```diff
@@ -0,0 +1,187 @@
+
+            // operational domain stats
+            operational_domain_stats op_domain_stats_gs{};
+            operational_domain_stats op_domain_stats_rs{};
+            operational_domain_stats op_domain_stats_ff{};
+            operational_domain_stats op_domain_stats_ct{};
+
+            // compute the operational domains
+            const auto op_domain_gs =
+                operational_domain_grid_search(lyt, truth_table, op_domain_params, &op_domain_stats_gs);
+            const auto op_domain_rs =
+                operational_domain_random_sampling(lyt, truth_table, 2500, op_domain_params, &op_domain_stats_rs);
+            const auto op_domain_ff =
+                operational_domain_flood_fill(lyt, truth_table, 250, op_domain_params, &op_domain_stats_ff);
+            const auto op_domain_ct =
+                operational_domain_contour_tracing(lyt, truth_table, 100, op_domain_params, &op_domain_stats_ct);
+
+            // write the operational domains to a CSV file
+            write_operational_domain(op_domain_gs,
+                                     fmt::format("{}operational_domain_grid_search_siqad_{}.csv", folder, gate),
+                                     write_op_domain_params);
+            write_operational_domain(op_domain_rs,
+                                     fmt::format("{}operational_domain_random_sampling_siqad_{}.csv", folder, gate),
+                                     write_op_domain_params);
+            write_operational_domain(op_domain_ff,
+                                     fmt::format("{}operational_domain_flood_fill_siqad_{}.csv", folder, gate),
+                                     write_op_domain_params);
+            write_operational_domain(op_domain_ct,
+                                     fmt::format("{}operational_domain_contour_tracing_siqad_{}.csv", folder, gate),
+                                     write_op_domain_params);
+
+            // update the total number of samples
+            total_samples_gs += op_domain_stats_gs.num_evaluated_parameter_combinations;
+            total_samples_rs += op_domain_stats_rs.num_evaluated_parameter_combinations;
+            total_samples_ff += op_domain_stats_ff.num_evaluated_parameter_combinations;
+            total_samples_ct += op_domain_stats_ct.num_evaluated_parameter_combinations;
+
+            // update the total number of simulator calls
+            total_sim_calls_gs += op_domain_stats_gs.num_simulator_invocations;
+            total_sim_calls_rs += op_domain_stats_rs.num_simulator_invocations;
+            total_sim_calls_ff += op_domain_stats_ff.num_simulator_invocations;
+            total_sim_calls_ct += op_domain_stats_ct.num_simulator_invocations;
+
+            // compute the operational percentages
+            const auto operational_percentage_gs =
+                static_cast<double>(op_domain_stats_gs.num_operational_parameter_combinations) /
+                op_domain_stats_gs.num_evaluated_parameter_combinations;
```

Comment:
My clang-tidy shows an issue because we convert ``std::size_t`` to ``double``

Reply:
Good catch!